packages/components/src/DataList/DataList.tsx

```diff
@@ -144,13 +144,20 @@ function InternalDataList({
   const showEmptyState = !initialLoading && data.length === 0;
 
   const shouldRenderLoadMoreTrigger = !initialLoading && !showEmptyState;
+  const shouldRenderTitleContainer =
+    initialLoading || title !== "" || totalCount !== null;
```

Comment:
a few thoughts:

**initialLoading**
I think I understand the rationale behind `initialLoading` being one of the conditions. if we're loading, we might not currently have our data & the corresponding `totalCount` but once the query resolves it could be a proper number. that flow makes sense to me.

the scenario that might be awkward is when we aren't providing the `totalCount` prop at all and have no intention of using it AKA it is `undefined`. if we also don't have a title, it'll render this section while loading, and then disappear once loading is done because we'll fail all the checks.

if we wanted to be as correct as possible, we'd wanna combine `initialLoading` with verifying at least one of the other elements is either there, or expected to be there. that would avoid adding an element we don't need.

**title**
since the type of title is either a `string` or `undefined`, `title !== ""` will result in the section rendering if you completely omit the `title` prop or directly set it to `undefined` so we'd definitely need to check for `undefined`.

I actually don't think we need to check or protect against an empty string. I'd rather leave that up to the consumer to handle in a more explicit way. if you don't want a title, we already provide a way to opt out so simply don't provide one. short circuiting on an empty string would effectively be a second way to achieve that same functionality, but in a less explicit way imo. this will tie into the next point around what `undefined` is meant to represent.

**totalCount**
this one has a surprising amount of hidden complexity 😅 

the type of totalCount is `number | undefined | null`, where omitting the prop is `undefined` so only checking against `null` won't quite get us what we want.

looking into it deeper, `null` is something we **do** want to let through because it represents a different flow & goes down a separate code path.

https://github.com/GetJobber/atlantis/blob/master/packages/components/src/DataList/components/DataListTotalCount/DataListTotalCount.tsx#L20-L22

inside the total count component we are treating `undefined` as the case where you have no total count at all. we show nothing. that will be one of the conditions we need to have out here.

conversely `null` is paired with `loading` to show the Glimmer on initial load only. I'm not sure I agree with that behavior, but I found a comment stating that it was intentional so I can follow up with that separately.

all that is to say, we should specifically checking that `totalCount !== undefined` out here. I'd imagine we can also refactor & remove the `undefined` check inside `DataListTotalCount` since there wouldn't be a way for us to hit that anymore.